ext/imap/php_imap.stub.php

```diff
@@ -5,251 +5,250 @@
+ * @param resource $imap
  * @alias imap_body
  */
-function imap_fetchtext($stream_id, int $msg_no, int $options = 0): string|false {}
+function imap_fetchtext($imap, int $message_num, int $flags = 0): string|false {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  * @return stdClass|false
  */
-function imap_bodystruct($stream_id, int $msg_no, string $section) {}
+function imap_bodystruct($imap, int $message_num, string $section) {}
 
-/** @param resource $stream_id */
-function imap_fetchbody($stream_id, int $msg_no, string $section, int $options = 0): string|false {}
+/** @param resource $imap */
+function imap_fetchbody($imap, int $message_num, string $section, int $flags = 0): string|false {}
 
-/** @param resource $stream_id */
-function imap_fetchmime($stream_id, int $msg_no, string $section, int $options = 0): string|false {}
+/** @param resource $imap */
+function imap_fetchmime($imap, int $message_num, string $section, int $flags = 0): string|false {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  * @param resource|string|int $file
  */
-function imap_savebody($stream_id, $file, int $msg_no, string $section = "", int $options = 0): bool {}
+function imap_savebody($imap, $file, int $message_num, string $section = "", int $flags = 0): bool {}
 
-/** @param resource $stream_id */
-function imap_fetchheader($stream_id, int $msg_no, int $options = 0): string|false {}
+/** @param resource $imap */
+function imap_fetchheader($imap, int $message_num, int $flags = 0): string|false {}
 
-/** @param resource $stream_id */
-function imap_fetchstructure($stream_id, int $msg_no, int $options = 0): stdClass|false {}
+/** @param resource $imap */
+function imap_fetchstructure($imap, int $message_num, int $flags = 0): stdClass|false {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  */
-function imap_gc($stream_id, int $flags): bool {}
+function imap_gc($imap, int $flags): bool {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  */
-function imap_expunge($stream_id): bool {}
+function imap_expunge($imap): bool {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  */
-function imap_delete($stream_id, string $msg_no, int $options = 0): bool {}
+function imap_delete($imap, string $message_num, int $flags = 0): bool {}
 
 /**
- * @param resource $stream_id
+ * @param resource $imap
  */
-function imap_undelete($stream_id, string $msg_no, int $options = 0): bool {}
+function imap_undelete($imap, string $message_num, int $flags = 0): bool {}
 
-/** @param resource $stream_id */
-function imap_check($stream_id): stdClass|false {}
+/** @param resource $imap */
+function imap_check($imap): stdClass|false {}
 
-/** @param resource $stream_id */
-function imap_listscan($stream_id, string $ref, string $pattern, string $content): array|false {}
+/** @param resource $imap */
+function imap_listscan($imap, string $reference, string $pattern, string $content): array|false {}
```

Comment:
I managed to track the `$ref` parameter all the way down to https://github.com/uw-imap/imap/blob/22f316e36dc00e445408f7a47750b7fa9c3ce369/src/c-client/imap4r1.c#L518, and I still have no clue what it does. The docs say:

> ref should normally be just the server specification as described in imap_open()

So maybe it is the "server part" of `$mailbox`? I have no idea.

Reply:
It looks to be another IMAP mailbox, but that doesn't make a lot of sense as we are already passing a mailbox.. I have no idea either.